src/runtime_src/core/pcie/emulation/hw_em/alveo_shim/shim.cxx

```diff
@@ -194,35 +194,43 @@ namespace xclhwemhal2 {
 
   int HwEmShim::parseLog()
   {
+    auto lstatus = 0;
     std::vector<std::string> myvector = {"SIM-IPC's external process can be connected to instance",
                                          "SystemC TLM functional mode",
-                                         "HLS_PRINT",
-                                         "Exiting xsim",
-                                         "FATAL_ERROR"};
+                                         "HLS_PRINT"};
 
     std::ifstream ifs(getSimPath() + "/simulate.log");
 
     if (ifs.is_open()) {
       std::string line;
       while (std::getline(ifs, line)) {
-        for (auto matchString : myvector) {
+        // If xsim is terminated by logging in simulate.log then call xclClose to properly clean all instances, threads.
+        if ( (std::string::npos != line.find("Exiting xsim")) || 
+             (std::string::npos != line.find("ERROR")) 
+           ) {
+                  std::cout << "SIMULATION EXITED" << std::endl;
+                  lstatus = -1;
+                  parsedMsgs.push_back(line);
+                  this->xclClose(true);                                               // Let's have a proper clean if xsim is NOT running
+        }
+        for (auto& matchString : myvector) {
           std::string::size_type index = line.find(matchString);
-          if (index != std::string::npos) {
-            if(std::find(parsedMsgs.begin(), parsedMsgs.end(), line) == parsedMsgs.end()) {
+          if (index == std::string::npos) {
+            continue;
+          }
+          else {
+            if(std::find(parsedMsgs.begin(), parsedMsgs.end(), line) != parsedMsgs.end()) {
+              continue;
+            }
+            else {
```

Comment:
```suggestion
            if (std::find(parsedMsgs.begin(), parsedMsgs.end(), line) != parsedMsgs.end())
              continue;

```

Reply:
extra braces and additional else conditions are removed. Implemented as per suggestion.